pkg/appsec/appsec.go

```diff
@@ -0,0 +1,114 @@
+package appsec
+
+import (
+	"encoding/json"
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/api/apiutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NewIntakeReverseProxy returns the AppSec Intake Proxy handler according to
+// the agent configuration.
+func NewIntakeReverseProxy(transport http.RoundTripper) (http.Handler, error) {
+	disabled := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Content-Type", "application/json")
+		w.WriteHeader(http.StatusNotImplemented)
+		json.NewEncoder(w).Encode("appsec api disabled")
```

Comment:
I think you can just use `http.Error` here.

Reply:
The intake api spec tells it returns a json value, even in case of error (explaining why), so I was thinking in aligning here.